File: solutions/fortran/raindrops/1/raindrops.f90
module raindrops
  implicit none
contains

  function convert(i)
    integer :: i
    character(len=:), allocatable :: convert
    convert=""
    if (mod(i,3) == 0) then
      convert = convert // "Pling"
    end if
    if (mod(i,5) == 0) then
      convert = convert // "Plang"
    end if
    if (mod(i,7) == 0) then
      convert = convert // "Plong"
    end if
    print *, "convert=", convert
    if (trim(convert) == "") then
      write(convert,'(i10)') i;
      convert=trim(convert)
    end if
    return

  end function convert

end module raindrops

File: solutions/julia/raindrops/1/raindrops.jl
function raindrops(number)
    rd=""
    if mod(number,3) == 0
        rd="Pling"
    end
    if mod(number,5) == 0
        rd=rd * "Plang"
    end
    if mod(number,7) == 0
        rd=rd * "Plong"
    end
    if rd==""
        rd=string(number)
    end
    return rd

end
